tests/python/pants_test/core_tasks/test_generate_pants_ini.py

```diff
@@ -0,0 +1,41 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import configparser
+import os
+from builtins import open
+from contextlib import contextmanager
+
+from pants.core_tasks.generate_pants_ini import GeneratePantsIni
+from pants.version import VERSION
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+class GeneratePantsIniIntegration(PantsRunIntegrationTest):
```

Comment:
Avoiding ITs when at all possible helps keep things fast and it will give you a temporary buildroot so you don't need to edit the real pants.ini.

```python
class GeneratePantsIniIntegration(ConsoleTaskTestBase):
  @classmethod
  def task_type(cls):
    return GeneratePantsIni

  def test_foo(self):
    self.execute_console_task()
    config = configparser.ConfigParser()
    config.read(os.path.join(self.buildroot, GeneratePantsIni.PANTS_INI))
    ...
```

Reply:
Ah good to know. I don't see something like `execute_task` in the API for `TaskTestBase`. Will wait to refactor this until we decide what to do about `ConsoleTask`.

Reply:
That would be `self.create_task(self.context()).execute()`.